File: 240228/국영수 순이지/korean-english-math-order.js
class Student {
    constructor(name, kor, eng, math) {
        this.name = name;
        this.kor = kor;
        this.eng = eng;
        this.math = math;
    }
}

const fs = require("fs");
let [n, ...data] = fs.readFileSync(0).toString().trim().split("\n");

const list = []
for(let d of data) {
    const [name, k, e, m] =  d.trim().split(" ");
    list.push(new Student(name, k, e, m))
}

list.sort((a,b) => (b.kor - a.kor) || (b.eng - a.eng) || (b.math - a.math))
list.forEach(s => {
    console.log(`${s.name} ${s.kor} ${s.eng} ${s.math}`)
})

File: 240228/국영수 순이지/korean-english-math-order.py
class Student: 
    def __init__(self, n, k, e, m):
        self.n = n
        self.k = k
        self.e = e
        self.m = m

n = int(input())
students = []

for _ in range(n):
    n, k, e, m = tuple(input().split())
    students.append(Student(n, int(k), int(e), int(m)))

students.sort(key = lambda x: (-x.k, -x.e, -x.m))

for s in students:
    print(s.n, s.k, s.e, s.m)